backend/app/models/resident.py

```diff
@@ -0,0 +1,37 @@
+from . import db
+from sqlalchemy import inspect
+from sqlalchemy.orm.properties import ColumnProperty
+
+class Resident(db.Model):
+    __tablename__ = "residents"
+    id = db.Column(db.Integer, primary_key=True, nullable=False)
+    initial = db.Column(db.String, nullable=False)
+    room_num = db.Column(db.Integer, nullable=False)
+    date_joined = db.Column(db.Date, nullable=False)
```

Comment:
if a resident has left SHOW, we should also store the date they left. It might not be implemented but its better to have the column in the db